app/views/hyku/api/v1/work/_work.json.jbuilder

```diff
@@ -21,8 +21,20 @@ json.cache! [@account, :works, work.id, work.solr_document[:_version_], work.mem
   json.cname @account.search_only? ? work.try(:solr_document)&.to_h&.dig("account_cname_tesim")&.first : @account.cname
   json.committee_member work.try(:solr_document)&.to_h&.dig('committee_member_tesim')
 
-  creator = work.creator.try(:first)
-  json.creator creator.present? ? JSON.parse(creator) : []
+  creator_hash = work.creator.try(:first)
+  creator_json = JSON.parse(creator_hash) if creator_hash.present?
+  if creator_json.present?
+    creator_json.each_with_index do |creator, index|
+      if creator["creator_institutional_email"].blank?
+        creator_json[index] = creator_json[index].except!("creator_institutional_email")
+        next
+      else
+        user = User.find_by(email: creator["creator_institutional_email"])
+        creator_json[index] = creator_json[index].except!("creator_institutional_email") unless user.present? && user.display_profile # Removes field if the user is not public/not found
+      end
+    end
+  end
+  json.creator creator_hash.present? ? creator_json : []
```

Comment:
This block is a little muddled. You're creating the `creator_hash` at the top, then checking the hash to create the json, processing the json then checking the hash again at the bottom. 

You shouldn't be doing this kind of querying inside a view really, the view should be as dumb as possible - I know that with Hyku its harder than it should be sometimes, but this kind of thing could be encapsulated inside a Helper and then just written as: 

```ruby
json.creator processed_creator(work.creator.try(:first))
```

Big sections of code like this can become hard to comprehend, so its best to not condense them down and remove the line breaks. If you space it out, it'll be easier for you to see where you can make improvements. 

Your variables are the wrong names. `creator_hash = work.creator.try(:first)` will return JSON, and `creator_json = JSON.parse(creator_hash)` will return a hash. 

The method `except!` changes the actual hash that you are on, so you dont need to assign that to another hash. So instead of: 
```ruby
creator_json[index] = creator_json[index].except!("creator_institutional_email")
```
You could just write 
```ruby
creator_json[index].except!("creator_institutional_email")
```
For example: 
```
2.7.4 :004 > h = {one: 1, two: 2}
 => {:one=>1, :two=>2}
2.7.4 :005 > h.except!(:one)
 => {:two=>2}
2.7.4 :006 > h
 => {:two=>2}
```
But you dont even need to do that as the `delete!` method will remove the requested key from the array. So, you can just use `creator_json[index].delete("creator_institutional_email")`. 

For example: 
```
2.7.4 :001 > h = {one: 1, two: 2}
 => {:one=>1, :two=>2}
2.7.4 :002 > h.delete(:one)
 => 1
2.7.4 :003 > h
 => {:two=>2}
```

You're using `creator_json.each_with_index do |creator, index|` and then accessing the `creator[index]`, but why not just use `map`?

I've not tested this, but adding this as  helper method would be preferable to the block you added to the view: 

```ruby
def processed_creator(creator_json)
  creator_hash = JSON.parse(creator_json || "{}") 

  return if creator_hash.blank?

  key = "creator_institutional_email"

  creator_hash.map do |creator|
    if creator[key].blank?
      creator.delete(key)
    else
      user = User.find_by(email: creator[key])
      creator_json[index].delete(key) unless user.present? && user.display_profile 
    end
  end
end
```